overlord/devicestate/handlers_gadget.go

```diff
@@ -210,11 +212,96 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 	return snapstate.FinishTaskWithRestart(t, state.DoneStatus, restart.RestartSystem, nil)
 }
 
-func (m *DeviceManager) updateGadgetCommandLine(t *state.Task, st *state.State, isUndo bool) (updated bool, err error) {
-	snapsup, err := snapstate.TaskSnapSetup(t)
+// fromSystemOption tells us if t was created when setting a system
+// option for the kernel command line.
+func fromSystemOption(t *state.Task) (bool, error) {
+	for _, param := range []string{"cmdline-append", "dangerous-cmdline-append"} {
+		var value string
+		err := t.Get(param, &value)
+		if err == nil {
+			return true, nil
+		}
+		if !errors.Is(err, state.ErrNoState) {
+			return false, err
+		}
```

Comment:
as we are ignoring values this can use t.Has

Reply:
Changed